packages/@react-native-windows/codegen/src/Cli.ts

```diff
@@ -151,31 +162,70 @@ function generate(
 */
 
   const generateNM2 = createNM2Generator({namespace: argv.namespace});
+  const generatorPropsH = require('react-native-tscodegen/lib/rncodegen/src/generators/components/GeneratePropsH')
+    .generate;
+  const generatorPropsCPP = require('react-native-tscodegen/lib/rncodegen/src/generators/components/GeneratePropsCPP')
+    .generate;
+  const generatorShadowNodeH = require('react-native-tscodegen/lib/rncodegen/src/generators/components/GenerateShadowNodeH')
+    .generate;
+  const generatorShadowNodeCPP = require('react-native-tscodegen/lib/rncodegen/src/generators/components/GenerateShadowNodeCPP')
+    .generate;
+  const generatorComponentDescriptorH = require('react-native-tscodegen/lib/rncodegen/src/generators/components/GenerateComponentDescriptorH')
+    .generate;
+  const generatorEventEmitterH = require('react-native-tscodegen/lib/rncodegen/src/generators/components/GenerateEventEmitterH')
+    .generate;
+
+  generatedModuleFiles.push(
+    ...generateNM2(libraryName, schema, moduleSpecName),
+  );
 
-  generatedFiles.push(...generateNM2(libraryName, schema, moduleSpecName));
+  generatedComponentFiles.push(
+    ...generatorPropsH(libraryName, schema, moduleSpecName),
+    ...generatorPropsCPP(libraryName, schema, moduleSpecName),
+    ...generatorShadowNodeH(libraryName, schema, moduleSpecName),
+    ...generatorShadowNodeCPP(libraryName, schema, moduleSpecName),
+    ...generatorComponentDescriptorH(libraryName, schema, moduleSpecName),
+    ...generatorEventEmitterH(libraryName, schema, moduleSpecName),
+  );
 
-  const filesToUpdate = new Map<string, string>([...generatedFiles]);
+  const moduleFilesToUpdate = new Map<string, string>([
+    ...generatedModuleFiles,
+  ]);
+  const componentFilesToUpdate = new Map<string, string>([
+    ...generatedComponentFiles,
+  ]);
 
   if (test === true) {
-    return checkFilesForChanges(filesToUpdate, outputDirectory);
+    return (
+      checkFilesForChanges(moduleFilesToUpdate, outputDirectory) &&
+      checkFilesForChanges(componentFilesToUpdate, componentOutputdir)
+    );
   }
 
-  return writeMapToFiles(filesToUpdate, outputDirectory);
+  return (
+    writeMapToFiles(moduleFilesToUpdate, outputDirectory) &&
+    writeMapToFiles(componentFilesToUpdate, componentOutputdir)
+  );
 }
 
 if ((argv.file && argv.files) || (!argv.file && !argv.files)) {
   console.error('You must specify either --file or --files.');
   process.exit(1);
 }
 
+if (!argv.libraryName) {
+  console.error('You must specify --libraryName');
+  process.exit(1);
+}
+
```

Comment:
`yargs` can do this for you. Just add `required: true` to the option above.

Reply:
For the bit highlighted above the section added they have some way to specify "requires one of these options" but don't remember the details.